src/libs/Network/MainQueue.js

```diff
@@ -84,7 +80,7 @@ function process() {
  * Non-cancellable requests like Log would not be cleared
  */
 function clear() {
-    networkRequestQueue = _.filter(networkRequestQueue, (request) => !request.data.canCancel);
+    _.filter(networkRequestQueue, (request) => request.command !== 'Log');
```

Comment:
Ok that's a good improvement to not cancel the logs. We still cancel the pending log requests here however.
https://github.com/Expensify/App/blob/c80e6cd0ffe44883f6eeaaa367f9a159eff351ea/src/libs/actions/SignInRedirect.js#L85

Reply:
How? Right above is where we clear the requests with `MainQueue.clear()` no?

Reply:
When we make a request, like a Log, with Network.post, we eventually call fetch and set a [signal](https://developer.mozilla.org/en-US/docs/Web/API/fetch#:~:text=Navigator.sendBeacon()%20API.-,signal,-An%20AbortSignal%20object) in the fetch options. The signal is an AbortSignal from an AbortController which we call cancellationController. It allows us to terminate pending requests.

When a user signs out and there's a Log request pending (in flight), we cancel those pending requests. So a log request could be cancelled before it reaches the backend I believe. [Previously](https://github.com/Expensify/App/pull/19675/files#diff-c49793c436ac19c8ea9e9bbcd940629ed1c7b18ec48657c0365873bf437d1b67L39-L43), we left the signal undefined if canCancel was false and so it would not be cancelled.